app/configurator/components/chart-configurator.tsx

```diff
@@ -146,30 +154,33 @@ export const DataFilterSelectGeneric = ({
     data?.dataCubesComponents?.dimensions.find(
       (d) => d.cubeIri === rawDimension.cubeIri
     ) ?? rawDimension;
-  const controls = dimension.isKeyDimension ? null : (
-    <Box sx={{ display: "flex", flexGrow: 1 }}>
-      <IconButton
-        disabled={disabled}
-        sx={{ ml: 2, p: 0 }}
-        onClick={onRemove}
-        size="small"
-      >
-        <Icon name="trash" size="16" />
-      </IconButton>
-    </Box>
-  );
 
   const sharedProps = {
     dimension,
     label: disableLabel ? null : (
       <>
         <OpenMetadataPanelWrapper component={dimension}>
-          {`${index + 1}. ${dimension.label}`}
+          <Typography variant="caption">
+            {`${index + 1}. ${dimension.label}`}
+          </Typography>
         </OpenMetadataPanelWrapper>
-        <InteractiveDataFilterToggle id={dimension.id} />
+        <Flex
+          sx={{
+            alignSelf: "flex-end",
+            alignItems: "center",
+            gap: 1,
+            marginRight: sideControls && dimension.isKeyDimension ? 7 : 0,
```

Comment:
I wonder if it would not be cleaner to manage the margin right via the parent using flex layout + gap.